broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -569,4 +593,28 @@ public void close() {
 
     super.close();
   }
+
+  private void onFailure() {
+    healthStatus.set(HealthStatus.UNHEALTHY);
+    if (failureListener != null) {
+      failureListener.onFailure();
+    }
+  }
+
+  private void onRecovered() {
+    healthStatus.set(HealthStatus.HEALTHY);
+    if (failureListener != null) {
+      failureListener.onRecovered();
+    }
+  }
+
+  @Override
+  public HealthStatus getHealthStatus() {
+    return healthStatus.get();
```

Comment:
Do we expect this to be called from a non-actor? If no then we can get rid of the `AtomicReference` and simply use `actor.call` and return an `ActorFuture<HealthStatus>`.

Reply:
For the first iteration, I thought it was easier to implement sync api so that implementation of health monitors is straight forward. But I also think that we would need async calls in future, especially for some modules to detect if they are stuck. Do you think we should immediately go with async api for `getHealthStatus`?

Reply:
We could keep it sync and join against the result for now as well, e.g.

```java
return actor.call(() -> healthStatus).join();
```